xds/src/main/java/com/linecorp/armeria/xds/BootstrapClusters.java

```diff
@@ -31,16 +29,22 @@
 final class BootstrapClusters implements SnapshotWatcher<ClusterSnapshot> {
 
     private final Map<String, ClusterSnapshot> clusterSnapshots = new HashMap<>();
+    private final Map<String, Cluster> clusters = new HashMap<>();
 
     BootstrapClusters(Bootstrap bootstrap, XdsBootstrapImpl xdsBootstrap) {
         if (bootstrap.hasStaticResources()) {
             final StaticResources staticResources = bootstrap.getStaticResources();
             for (Cluster cluster : staticResources.getClustersList()) {
-                checkArgument(cluster.hasLoadAssignment(),
-                              "Only fully static configurations are allowed for bootstrap clusters. " +
-                              "Violating cluster is %s", cluster);
-                // no need to clean this cluster up since it is fully static
-                StaticResourceUtils.staticCluster(xdsBootstrap, cluster.getName(), this, cluster);
+                if (cluster.hasLoadAssignment()) {
+                    // no need to clean this cluster up since it is fully static
+                    StaticResourceUtils.staticCluster(xdsBootstrap, cluster.getName(), this, cluster);
+                    clusters.put(cluster.getName(), cluster);
+                } else if (cluster.hasEdsClusterConfig()) {
+                    clusters.put(cluster.getName(), cluster);
+                } else {
+                    throw new IllegalArgumentException(
+                            "Cluster must have a load assignment or EDS cluster config.");
+                }
```

Comment:
nit; I think it's perfectly legal for clusters to not have a load assignment (hence, `ClusterSnapshot#endpointSnapshot` is nullable)

Would it make sense to just always add to `clusters`?

```suggestion
                }
                clusters.put(cluster.getName(), cluster);
```

Reply:
Yep, make sense. 😉